### PR TITLE
Refactor const.rb - freeze

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

Ruby 2.3 implemented the magic comment `# frozen_string_literal: true`.  The below shows one interesting result, `join`:
```ruby
v1 = "Frozen?"
v2 = %w[HEAD GET POST PUT DELETE OPTIONS TRACE PATCH].freeze
v3 = v2.join ' '
v4 =  {501 => "HTTP/1.1 501 Not Implemented\r\n\r\n",
       503 => "HTTP/1.1 503 Service Unavailable\r\n\r\nBUSY"}

puts v1.frozen?      # -> true
puts v2[0].frozen?   # -> true
puts v3.frozen?      # -> false
puts v4[501].frozen? # -> true
```

So, remove `.freeze` when not needed in const.rb.  Also, remove extra constants (`GET POST PUT DELETE OPTIONS TRACE PATCH`) that were added in https://github.com/puma/puma/commit/1b6b8adfaeb4.

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [ ] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/const.rb [hunk 1]
```diff
@@ -112,28 +111,28 @@ module Const
     WRITE_TIMEOUT = 10
 
     # The original URI requested by the client.
-    REQUEST_URI= 'REQUEST_URI'.freeze
-    REQUEST_PATH = 'REQUEST_PATH'.freeze
-    QUERY_STRING = 'QUERY_STRING'.freeze
-    CONTENT_LENGTH = "CONTENT_LENGTH".freeze
+    REQUEST_URI= 'REQUEST_URI'
+    REQUEST_PATH = 'REQUEST_PATH'
+    QUERY_STRING = 'QUERY_STRING'
```
Comment (dentarg) [inline lib/puma/const.rb:116]: Suggestion: now that we are touching these lines, let's take the opportunity to make things a tiny bit more consistent by using `"` over `'`

```suggestion
    REQUEST_URI= "REQUEST_URI"
    REQUEST_PATH = "REQUEST_PATH"
    QUERY_STRING = "QUERY_STRING"
```

File: lib/puma/const.rb [hunk 2]
```diff
@@ -112,28 +111,28 @@ module Const
...
+    CONTENT_LENGTH = "CONTENT_LENGTH"
 
-    PATH_INFO = 'PATH_INFO'.freeze
+    PATH_INFO = 'PATH_INFO'
```
Comment (dentarg) [inline lib/puma/const.rb:119]:
```suggestion
    PATH_INFO = "PATH_INFO"
```

**Review by dentarg — COMMENTED** [review]
Usage of `'` comes from `HTTP_STATUS_CODES` and `PUMA_SERVER_STRING`, otherwise this file uses `"`, address that if you want or save it for later (in general I dislike doing changes like this because it makes `git blame` harder to use but if we are touching them anyways...)

**Comment by MSP-Greg** [comment]
@dentarg

For you, I changed all `'` to `"` in Puma::Const.  Left `HTTP_STATUS_CODES` (in the Puma namespace) the same.

 Personally, I prefer `'`...

**Review by dentarg — APPROVED** [review]

**Comment by MSP-Greg** [comment]
@dentarg Thaks for the review.

I've looked at it before, but (of course) now that 6.0.0 is released, I'm really looking at it.

There are a lot of constants that could be moved, as they're only used in one file.  For instance, the comment for `HTTP_STATUS_CODES` states "These are used so frequently that they are placed directly in Puma".  Today, the only file (including tests) I find them in is request.rb.  Same for some of the constant 'groups' in Puma::Const.

Any thoughts?

**Comment by dentarg** [comment]
No real thoughts on that other than it sounds like a different PR :)

**Comment by MSP-Greg** [comment]
> it sounds like a different PR

Of course...

**Comment by MSP-Greg** [comment]
@nateberkopec

Thoughts?  Just checking, as this does remove some constants added to 6.0.0.  See updated first post.

**Review by francois-ferrandis — APPROVED** [review]
Seems like a good idea, I was wondering why all these freezes were still in place! :slightly_smiling_face:

**Pull request merged**